### DISCUSSION TITLE
Listening for pending payments/transfers to an address on supported networks

### DISCUSSION THREAD

**Discussion by bjuretic** [blockscout/blockscout#10648] [General]
Hi guys,

I am trying to figure out how to "listen", using GraphQL, REST API or some other programatic way for incoming transfers/payments  into a particular address.  So imagine for example a service like BitPay, which needs to know as soon as possible that a payment arrived into their wallet address so they can inform their merchant, that kind of a use case. Yes, I can poll the addresses/:address_uid/transactions but there must be a better way, as there is obviously some event which gets sent via web socket to the UI when browsing a particular address, which informs the user that there are new transactions (since the time he opened the page). I am basically trying to find something like that, which I can use directly, without the UI (but with an API token).

Thanks in advance for the replies.

**Comment by sl1depengwyn** [comment]
You're right, there is actually a websocket that sends new address transactions, but we do not have any docs on how to use it afaik, so I can suggest you to look through https://github.com/blockscout/frontend source and check how it used there
